### PR TITLE
DRA func_800F2120

### PR DESCRIPTION
**Pull request by cram0**
Decompiles func_800F2120

PS : Commit eef6d63f88ada8391f359b18ecf542d81d4c0dfd removes an unnecessary parameter in another function that I decompiled earlier, which shouldnt exist in the first place so I fixed it, I didn't want to make a second PR just for it.

### CODE REVIEW

File: src/dra/5087C.c [hunk 1]
```diff
@@ -524,13 +524,41 @@ void func_800F2014(void) {
...
+    u8 currMapRect;
+
+    func_800F1A3C(g_StageId & STAGE_INVERTEDCASTLE_FLAG);
+    ClearImage(&g_Vram.D_800ACDE8, 0U, 0U, 0U);
+    DrawSync(0);
+
+    for (y = 0; y < 64; y++) {
+        for (x = 0; x < 64; x++) {
+            idx = (x >> 2) + (y * 16);
+            subMap = 3 << ((3 - (x & 3)) * 2);
```
Comment (sozud) [inline src/dra/5087C.c:547]: Some docs:

```
            // sequence of 2 bit masks: 0xC0, 0x30, 0x0C, 0x03
            // 0b11000000, 0b110000, 0b1100, 0b11
            subMap = 3 << ((3 - (x & 3)) * 2);

            if (g_StageId & STAGE_INVERTEDCASTLE_FLAG) {
                idx += 0x400;
            }
            currMapRect = g_CastleMap[idx];
            // check for odd or even bits (0x55 / 0xAA)
            if (currMapRect & 0x55 & subMap) {
                func_800F1B08(x, y, 0);
                func_800F1EB0(x, y, 0xFFFF);
            } else if (currMapRect & 0xaa & subMap) {
                func_800F1B08(x, y, 1);
            }
            ```

File: src/dra/5087C.c [hunk 2]
```diff
@@ -524,13 +524,41 @@ void func_800F2014(void) {
...
-INCLUDE_ASM("dra/nonmatchings/5087C", func_800F2120);
+void func_800F2120(void) {
+    s32 x;
+    s32 y;
+    s32 subMap;
+    s32 idx;
+    u8 currMapRect;
+
+    func_800F1A3C(g_StageId & STAGE_INVERTEDCASTLE_FLAG);
+    ClearImage(&g_Vram.D_800ACDE8, 0U, 0U, 0U);
```
Comment (sozud) [inline src/dra/5087C.c:541]: The `U`s can be removed

File: src/dra/5087C.c (hunk 1 above)
Comment (cram0) [reply src/dra/5087C.c:547]: Noted, thank you !

File: src/dra/5087C.c [hunk 3]
```diff
@@ -524,13 +524,41 @@ void func_800F2014(void) {
...
+    u8 currMapRect;
+
+    func_800F1A3C(g_StageId & STAGE_INVERTEDCASTLE_FLAG);
+    ClearImage(&g_Vram.D_800ACDE8, 0, 0, 0);
+    DrawSync(0);
+
+    for (y = 0; y < 64; y++) {
+        for (x = 0; x < 64; x++) {
+            idx = (x >> 2) + (y * 16);
+            subMap = 3 << ((3 - (x & 3)) * 2);
```
Comment (sozud) [inline src/dra/5087C.c:547]: Commenting again to make this more clear:
```
            // sequence of 2 bit masks: 0xC0, 0x30, 0x0C, 0x03
            // 0b11000000, 0b110000, 0b1100, 0b11
 ```
 Could you add those two comments above `subMap = 3 << ((3 - (x & 3)) * 2);` to document what it does?

File: src/dra/5087C.c [hunk 4]
```diff
@@ -524,13 +524,41 @@ void func_800F2014(void) {
...
+
+            if (g_StageId & STAGE_INVERTEDCASTLE_FLAG) {
+                idx += 0x400;
+            }
+            currMapRect = g_CastleMap[idx];
+            if (currMapRect & 0x55 & subMap) {
```
Comment (sozud) [inline src/dra/5087C.c:553]: Could you add a comment like:

`0x55 and 0xAA are masks for even and odd bits. e.g. 0x55 is 0b1010101`

File: src/dra/5087C.c (hunk 3 above)
Comment (cram0) [reply src/dra/5087C.c:549]: Done, thank you for the clarification

File: src/dra/5087C.c (hunk 4 above)
Comment (cram0) [reply src/dra/5087C.c:557]: Done, thank you for the clarification

**Pull request merged**